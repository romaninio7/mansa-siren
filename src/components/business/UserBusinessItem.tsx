import * as React from 'react';
import { IUserBusinessItem } from 'interfaces';

const UserBusinessItem = ({
  businessValue,
  businessData,
}: IUserBusinessItem): JSX.Element => {
  return (
    <div className='item'>
      <div className={`ui label large ${businessData[0]}`}>
        <i className={`id icon ${businessData[1]}`}></i>
        <span className='business-name'>{businessData[2]}: </span>
      </div>
      <span className={businessData[3]}>{businessValue}</span>
    </div>
  );
};

export default UserBusinessItem;
