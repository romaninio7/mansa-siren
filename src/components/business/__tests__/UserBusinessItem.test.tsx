import * as React from 'react';
import UserBusinessItem from 'components/business/UserBusinessItem';
import { mount } from 'enzyme';

let wrapper;

describe('UserBusinessItem component rendering', () => {
  beforeEach(() => {
    wrapper = mount(
      <UserBusinessItem
        businessValue='879878949'
        businessData={['blue', 'briefcase', 'Siret', 'siret']}
      />
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('has a valid snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('shows a valid business data', () => {
    expect(wrapper.find('.siret').text()).toEqual('879878949');
    expect(wrapper.exists('.blue')).toBeTruthy();
    expect(wrapper.exists('.briefcase')).toBeTruthy();
    expect(wrapper.find('.business-name').text()).toEqual('Siret: ');
  });
});
