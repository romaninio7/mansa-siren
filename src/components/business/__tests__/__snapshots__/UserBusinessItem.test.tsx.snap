// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBusinessItem component rendering has a valid snapshot 1`] = `
<UserBusinessItem
  businessData={
    Array [
      "blue",
      "briefcase",
      "Siret",
      "siret",
    ]
  }
  businessValue="879878949"
>
  <div
    className="item"
  >
    <div
      className="ui label large blue"
    >
      <i
        className="id icon briefcase"
      />
      <span
        className="business-name"
      >
        Siret
        : 
      </span>
    </div>
    <span
      className="siret"
    >
      879878949
    </span>
  </div>
</UserBusinessItem>
`;
